### PR TITLE
tidyselect syntax when removing tables

### PR DESCRIPTION
**Pull request by TSchiefer**
- select helpers available for `cdm_rm_tbl()`
- helper function for carefully selecting tables from a `dm`

### CODE REVIEW

File: R/add-tbl.R [hunk 1]
```diff
@@ -83,3 +78,18 @@ check_new_tbls <- function(dm, tbls) {
   }
 }
 
+dm_try_tables <- function(quo, table_names) {
```
Comment (krlmlr) [inline R/add-tbl.R]:
```suggestion
dm_try_tables <- function(code, table_names) {
```

and later

```r
quo <- enquo(code)
```

But we don't need the `quo` later. Deferred evaluation works with plain R, see https://github.com/tidyverse/tibble/blob/master/R/msg.R#L297 for a similar implementation. Can you please model after this?

File: R/add-tbl.R [hunk 2]
```diff
@@ -94,3 +89,18 @@ check_new_tbls <- function(dm, tbls) {
   }
 }
 
+dm_try_tables <- function(code, table_names) {
+  selected <- tryCatch(
+    force(code),
+    error = identity)
```
Comment (krlmlr) [inline R/add-tbl.R]:
```suggestion
    function(e) {
      ...
    }
  )
```

File: R/add-tbl.R [hunk 3]
```diff
@@ -70,21 +70,16 @@ cdm_add_tbl_impl <- function(dm, tbls, table_name, filters = vctrs::list_of(new_
 #'
 #' @param dm A [`dm`] object.
 #' @param ... One or more unquoted table names to remove from the `dm`.
+#' `tidyselect` functions are supported.
 #'
 #' @export
 cdm_rm_tbl <- function(dm, ...) {
   check_dm(dm)
+  selected <- dm_try_tables(setdiff(src_tbls(dm), tidyselect::vars_select(src_tbls(dm), ...)), src_tbls(dm))
```
[the remark this answers is not in the document]
Comment (TSchiefer) [reply R/add-tbl.R]: I looked a bit more into it now. There are 2 different types of errors that do occur:
1. error of class `simpleError` when called from `cdm_flatten_to_tbl()` (call is `tidyselect::vars_select(vars, ...)`) and one or more of `...` not in `vars`
2. error of class `rlang_error` when called from `cdm_rm_tbl()` (call is `setdiff(src_tbls(dm), tidyselect::vars_select(src_tbls(dm), ...))`) and one or more of `...` not in `src_tbls(dm)`

I am checking for both of them now and build a fitting error message. Only problem: what if a different `simpleError` or `rlang_error` occurs?

[the remark this answers is not in the document]
Comment (krlmlr) [reply R/add-tbl.R]: Filed in #200.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**